samcli/commands/deploy/utils.py

```diff
@@ -25,15 +27,17 @@ def print_deploy_args(stack_name, s3_bucket, region, capabilities, parameter_ove
     :param s3_bucket: Name of s3 bucket used for packaging code artifacts
     :param region: Name of region to which the current sam/cloudformation stack will be deployed to.
     :param capabilities: Corresponding IAM capabilities to be used during the stack deploy.
+    :param template_params: Parameters from template, need to hide "NoEcho" paramters
     :param parameter_overrides: Cloudformation parameter overrides to be supplied based on the stack's template
     :param confirm_changeset: Prompt for changeset to be confirmed before going ahead with the deploy.
     :return:
     """
 
     _parameters = parameter_overrides.copy()
     for key, value in _parameters.items():
-        if isinstance(value, dict):
-            _parameters[key] = value.get("Value", value) if not value.get("Hidden") else "*" * len(value.get("Value"))
```

Comment:
I think you maintain the same logic.
In your change, you neglect if the Parameter value is a dict not just a simple string

Reply:
Thanks. This has been updated to use only 5 stars in case value is a dict.